shared/src/commonTest/kotlin/co/nimblehq/avishek/phong/kmmic/domain.usecase/GetSurveyDetailUseCaseTest.kt

```diff
@@ -0,0 +1,63 @@
+package co.nimblehq.avishek.phong.kmmic.domain.usecase
+
+import co.nimblehq.avishek.phong.kmmic.data.remote.model.toSurvey
+import co.nimblehq.avishek.phong.kmmic.domain.repository.SurveyRepository
+import co.nimblehq.avishek.phong.kmmic.helper.MockUtil
+import io.kotest.matchers.shouldBe
+import io.mockative.Mock
+import io.mockative.any
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+
+@ExperimentalCoroutinesApi
+class GetSurveyDetailUseCaseTest {
+
+    @Mock
+    private val mockRepository = mock(classOf<SurveyRepository>())
+
+    private lateinit var useCase: GetSurveyDetailUseCase
+
+    @BeforeTest
+    fun setUp() {
+        useCase = GetSurveyDetailUseCaseImpl(mockRepository)
+    }
+
+    @Test
+    fun `when getSurvey is called successfully, it returns a survey`() = runTest {
+        given(mockRepository)
+            .function(mockRepository::getSurvey)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(MockUtil.mockSurveyApiModel.toSurvey()))
```

Comment:
We can keep `MockUtil.mockSurveyApiModel.toSurvey()` in a property, e.g.,
```Kotlin
val expectedSurvey = MockUtil.mockSurveyApiModel.toSurvey()
// ...
useCase("survey_id").first() shouldBe expectedSurvey
```

Reply:
solved in [05a48e0](https://github.com/nimblehq/avishek-phong-kmm-ic/pull/119/commits/05a48e0eab1776cafd561a4724b5775f2122792c)